src/ModelSingleton.py

```diff
@@ -172,17 +173,22 @@ def __call__(self, query_text):
 
 
 class SapbertModelWrapper(ModelWrapper):
-    def __init__(self, model_path, all_reps_path, all_reps_ids_path):
+    """
+        host: "http://localhost:9200"
+    username: "elastic"
+    password: ""
+    index: "sap_index"
+    """
+    def __init__(self, model_path, elastic_search_config):
         """ Initializes NLP Model"""
         super(SapbertModelWrapper, self).__init__()
         self.tokenizer = AutoTokenizer.from_pretrained(model_path)
         self.model = AutoModel.from_pretrained(model_path).cuda(0)
-        self.all_reps_emb = np.load(all_reps_path)
-        all_reps_name_ids = pd.read_csv(all_reps_ids_path, header=0, dtype=str)
-        self.all_reps_names = all_reps_name_ids['Name']
-        self.all_reps_ids = all_reps_name_ids['ID']
+        self.elastic_client = SAPElastic(
+            **elastic_search_config
+        )
```

Comment:
I am wondering whether we need to have a configuration variable to indicate whether it uses elastic search for prediction or not. Then we can keep both elastic search configuration initialization as shown here as well as the old non-elastic search configurations, i.e., loading ground truth data directly into memory. This way the same code base can support both setups, i.e., babel sapbert using elastic search and pubmed sapbert using in-memory nearest neighbor search.